rfcs/20220307-monitoring_across_the_stack.md

```diff
@@ -0,0 +1,361 @@
+* Alternative to the CloudWatch exporter is to implement the Node exporter: https://github.com/prometheus/node_exporter
+    * This has the added benefit of being portable across cloud providers vs CloudWatch being AWS services only.
+    * Does not cover ECS in an obvious way; could we run the node exporter in the container? Would that produce meaningful information?
+* Additional alternative to CloudWatch exporter is EC2 exporter: https://github.com/jmal98/ec2-exporter
+* Airflow exporter: https://github.com/epoch8/airflow-exporter
+
+### Grafana alerting integrations
+
+Grafana has built in alerting since v4.0. It works upstream from provider alertmanagers, so if we wanted to consume alerting rules from Prometheus we could build rules in Prometheus and then consume them in Grafana and alert on them.
+
+[Here is the list of Grafana "contact points" for alerting.](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/contact-points/) A contact point is the notification channels that Grafana can send alerts to.
+
+The most relevant for our usage are probably Slack, Webhook, and Email. Optionally we could incorporate PagerDuty depending on the team's tolerance and how we decide to organize expectations around responding to alerts.
+
+My preference is for us to primarily rely on email and in particular to use email lists to allow individuals to subscribe/unsubscribe at will from alarm streams and provide an easy historical public record of alarms. We could use something like [GNU Mailman](https://list.org/) to [host this ourselves](https://github.com/maxking/docker-mailman) using FARGATE.
+
+Email also makes it easy to compartmentalize alarms per category, it makes it easy for individuals to subscribe to specific streams of alarms by using different email prefixes per category. For example, each service could have its own mailing list.
+
+We can also integrate Slack for good measure and have all alarms sent to the relevant channels via Slack incoming Webhook. It's very easy to configure the Slack integration.
+
+The general purpose Webhook integration is useful for a future where we migrate away from Slack to Matrix (should that ever happen).
+
+#### Runbooks
+
+Runbooks are the expected steps for addressing a particular alarm. As I mentioned elsewhere, alarms should be actionable. If they're not actionable then they're just noise and should be demoted to a warning until they're more accurate (less false positives). The _actions_ you take for any given alarm are directed by an alarm's accomanying runbook.
+
+All alarms should have a relevant runbook linked in the alarm body. Ideally these would just be flat Markdown files but it'd be best if they could be hosted in the Openverse handbook on WordPress.org as well for redundancy and accessibility.
+
```

Comment:
+1 I'm glad you called this out in particular. This really exposes which alarms are meaningful and actionable. I'd add that it's also important for making it easier to distribute the work of _addressing_ alarms 💯

Reply:
This section is fantastic.